Tasks/BatchScript/BatchScript.ps1

```diff
@@ -21,26 +21,16 @@ Write-Verbose "modifyEnvironment (converted) = $allowModifyEnvironment"
 # Check for file existence
 if ([System.IO.File]::Exists($filename))
 {
-    $currentLocation = Get-Location
-    Write-Verbose "Current working folder: $currentLocation"
-    if ($workingFolder)
-    {
-        Write-Verbose "Changing the current working folder to: $workingFolder"
-        Set-Location -Path $workingFolder
-        $currentLocation = Get-Location
-        Write-Verbose "New working folder: $currentLocation"
-    }
-
     if ($allowModifyEnvironment)
     {
         Write-Verbose "Invoking script $filename with AllowScriptToChangeEnvironment flag set"
         if ($arguments)
         {
-            Invoke-BatchScript $filename -AllowScriptToChangeEnvironment -Arguments $arguments
+            Invoke-BatchScript $filename -AllowScriptToChangeEnvironment -Arguments $arguments -WorkingFolder $workingFolder
```

Comment:
We probably still need some code at the top to check working folder. If it is null/empty passing the argument will fail with a binding error.